src/api/spec/controllers/webui/staging/excluded_requests_controller_spec.rb

```diff
@@ -0,0 +1,71 @@
+require 'rails_helper'
+
+RSpec.describe Webui::Staging::ExcludedRequestsController, type: :controller do
+  let(:user) { create(:confirmed_user, login: 'tom') }
+  let(:project) { user.home_project }
+  let(:factory) { create(:project, name: 'openSUSE:Factory') }
+  let(:staging_workflow) { project.create_staging }
+
+  let(:source_package) { create(:package) }
+  let(:target_package) { create(:package, name: 'target_package', project: factory) }
+  let(:bs_request) do
+    create(:bs_request_with_submit_action,
+           target_project: target_package.project.name,
+           target_package: target_package.name,
+           source_project: source_package.project.name,
+           source_package: source_package.name)
+  end
+
+  before do
+    login(user)
+  end
+
+  describe '#create' do
+    let(:description) { Faker::Lorem.sentence }
+
+    context 'success' do
+      before do
+        post :create, params: { staging_workflow_id: staging_workflow, excluded_request: { bs_request_id: bs_request.id, description: description } }
+      end
+
+      it { expect(Staging::RequestExclusion.count).to eq(1) }
+      it { expect(staging_workflow.request_exclusions.first.description).to eq(description) }
+      it { expect(response).to redirect_to(staging_workflow_excluded_requests_path(staging_workflow)) }
+      it { expect(flash[:success]).not_to be_nil }
+    end
+
+    context "doesn't exist" do
+      before do
+        allow_any_instance_of(Staging::RequestExclusion).to receive(:save).and_return(false)
+        post :create, params: { staging_workflow_id: staging_workflow, excluded_request: { bs_request_id: bs_request.id, description: description } }
```

Comment:
once you use the number in the controller, then you can remove the `.id` as we have a param method in BsRequest which does the magic for you :wink: